libreswan: Unbreak the package

iblech opened the pull request:
On current unstable, libreswan compiles but then doesn't run because of an ABI
change in NSS.

There is an easy fix which is to land in the next version of libreswan, see
https://github.com/libreswan/libreswan/issues/334; in the meantime, we can copy
this fix so that libreswan works again on unstable.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

iblech committed:
libreswan: Unbreak the package

On current unstable, libreswan compiles but then doesn't run because of an ABI
change in NSS.

There is an easy fix which is to land in the next version of libreswan, see
https://github.com/libreswan/libreswan/issues/334; in the meantime, we can copy
this fix so that libreswan works again on unstable.

SuperSandro2000 commented:
While we are at it:

I tried building libreswan on darwin and got the following error:

```
$ nix-build -A libreswan
error: Package ‘iproute2-5.8.0’ in /Users/user/nixpkgs/pkgs/os-specific/linux/iproute/default.nix:48 is not supported on ‘x86_64-darwin’, refusing to evaluate.

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.

(use '--show-trace' to show detailed location information)
```

We should probably remove the darwin platform:

```diff
-    platforms = platforms.linux ++ platforms.darwin ++ platforms.freebsd;
+    platforms = platforms.linux ++ platforms.freebsd;
```

iblech commented:
Good catch, thank you @SuperSandro2000! Do you have a way to test this later? How come that libreswan ever worked on darwin? (I don't run darwin and also don't really have any clue about libreswan.)

SuperSandro2000 commented:
> Do you have a way to test this later?

I can't really fiddle with network right now. Sorry.

> How come that libreswan ever worked on darwin?

This is like this since 2016 https://github.com/nixos/nixpkgs/blob/69d8cb4a6bf85b2a6546a20a9e8b9c9c0f2943ec/pkgs/tools/networking/libreswan/default.nix and probably no one ever tested it on darwin.

@r-burns Can maybe you help out?

iblech commented:
Okay, then I'll push a commit to remove darwin for now. Thank you for your quick response. :-) If r-burns or someone else has time to look into darwin, we can always reactivate it. Right not it is broken on darwin either way, so it's not a regression to remove darwin from the supported platforms.

iblech committed:
libreswan: remove darwin from list of supported platforms

See discussion at https://github.com/NixOS/nixpkgs/pull/101543. Currently,
this build depends on iproute2 which is not available on darwin. This might be
fixed in the future by employing iproute2mac.

r-burns reviewed (approved):
Thanks for the ping, yeah I think removing darwin from platforms is the right call. There are a ton of linux-only dependencies beyond iproute2, and the upstream project only claims to support linux.

Builds for me on linux. Is there a command I can run to verify the fix is working?

iblech commented:
Thank you for the quick review!

Unfortunately, I know of no quick command for testing. You need to enable the libreswan service with some dummy configuration.

SuperSandro2000 commented:
Result of `nixpkgs-review pr 101543` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>hash-slinger</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libreswan</li>
  </ul>
</details>

iblech commented:
Can this pull request be merged? For as long as

iblech commented:
Awesome, thank you @SuperSandro2000!
